### ISSUE TITLE
DNS not working when using VPN

### ISSUE THREAD

**Issue by pgerber** [n8fr8/orbot#175]
After updating to Orbot 16.0.3-BETA-1-tor-0.3.4.8, DNS is no longer working. Based on logcat there is something wrong with the setup of pdnsd.

Logcat:
```
09-16 14:29:50.291  2580  2580 D OrbotVpnService: starting OrbotVPNService service!
09-16 14:29:50.312  1748  1858 I ActivityManager: Displayed org.torproject.android/.vpn.VPNEnableActivity: +223ms 
(total +2m51s261ms)
09-16 14:29:50.318  2580  9750 I OrbotVpnService: PDNSD: 1
09-16 14:29:50.318  2580  9750 D OrbotVpnService: pdnsd: /data/data/org.torproject.android/app_bin/pdnsd[1]: syntax error: unexpected ')'
```

[full log](https://github.com/n8fr8/orbot/files/2386588/log.txt)

Content of `/data/data/org.torproject.android/app_bin/pdnsd.conf` looks fine though:

```
global { perm_cache=0; cache_dir=/data/data/org.torproject.android/app_bin; server_port = 8091; server_ip = 0.0.0.0; query_method=udp_only; min_ttl=15m; max_ttl=1w; timeout=10; daemon=on; pid_file=/data/data/org.torproject.android/app_bin/pdnsd.pid; } server { label= upstream; ip = 127.0.0.1; port = 5400; uptest = none; } rr { name=localhost; reverse=on; a=127.0.0.1; owner=localhost; soa=localhost,root.localhost,42,86400,900,86400,86400; }
```

Affected devices:
* Blackphone 2, Android 6.0.1
* Samsung Galaxy 4, LineageOS 15.1 (Android 8.1.0)

**Comment by Unpublished** [comment]
A log with this commit applied https://github.com/Unpublished/orbot/commit/5590fdc6958951990b615d8347f133b1c81b743f would be interesting.

As it doesn't seem to be caused by pdnsd itself but rather the command to execute it (just a guess).

If you don't want to compile yourself, maybe @n8fr8 can provide a test build if he doesn't know a fix yet.

**Comment by pgerber** [comment]
Closing as I can no longer reproduce this with the newly published Orbot 16.0.3-BETA-2-tor-0.3.4.8. Orbot works fine now on both of my devices.

Thanks @Unpublished for looking into the issue.

**Comment by n8fr8** [comment]
Glad to hear. We made some specific fixes for this (recompiled pdnsd DNS daemon that we use with the VPN service)
